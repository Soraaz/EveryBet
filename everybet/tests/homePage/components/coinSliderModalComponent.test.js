import React from "react";
import renderer from "react-test-renderer";
import {shallow} from "enzyme";

import CoinSliderModalComponent from "../../../src/homePage/components/coinSliderModalComponent";
import User from "../../../src/classes/user";
import {TouchableOpacity} from "react-native";

describe("CoinSliderModalComponent", () => {
    describe("rendering", () => {
        it("Should render correctly the component", () => {
            const tree = renderer
                .create(
                    <CoinSliderModalComponent
                        showModal={true}
                        switchModalVisibility={() => {}}
                        confirmAnswerFunction={() => {}}
                        selectedAnswer={{id: 0, title: "test selectedAnswer"}}
                        bet={{betId: 0, name: "test"}}
                    />
                )
                .toJSON();
            expect(tree).to.matchSnapshot();
        });
    });

    describe("functionalities", () => {
        let switchModalVisibility;
        let confirmAnswerFunction;
        let wrapper;
        let instance;

        beforeEach(() => {
            User.coins = 999999999;
            switchModalVisibility = sinon.spy();
            confirmAnswerFunction = sinon.spy();
            wrapper = shallow(
                <CoinSliderModalComponent
                    showModal={true}
                    switchModalVisibility={switchModalVisibility}
                    confirmAnswerFunction={confirmAnswerFunction}
                    selectedAnswer={{id: 0, title: "test selectedAnswer"}}
                    bet={{betId: 0, name: "test"}}
                />
            );
            instance = wrapper.instance();
        });

        it("Should replace undefined textInput by 1", () => {
            expect(instance.state.betCoins).to.be.equal("1");
            instance._onTextChange(undefined);
            expect(instance.state.betCoins).to.be.equal("1");
        });

        it("Should replace wrong textInput by 1", () => {
            expect(instance.state.betCoins).to.be.equal("1");
            instance._onTextChange("wrong input");
            expect(instance.state.betCoins).to.be.equal("1");
        });

        it("Should replace 0 by 1", () => {
            expect(instance.state.betCoins).to.be.equal("1");
            instance._onTextChange("0");
            expect(instance.state.betCoins).to.be.equal("1");
        });

        it("Should change the value of textInput", () => {
            expect(instance.state.betCoins).to.be.equal("1");
            instance._onTextChange("123456");
            expect(instance.state.betCoins).to.be.equal("123456");
        });

        it("Should replace textInput by the user's coin if textInput > user's coin", () => {
            User.coins = 42;
            expect(instance.state.betCoins).to.be.equal("1");
            instance._onTextChange("69");
            expect(instance.state.betCoins).to.be.equal("42");
        });

        it("Should call switchModalVisibility callback when pressing outside the modal", () => {
            expect(switchModalVisibility).to.be.not.called;
            wrapper.find(TouchableOpacity).first().props().onPress();
            expect(switchModalVisibility).to.be.calledOnce;
        });

        it("Should call switchModalVisibility callback when pressing the \"Annuler\" button", () => {
            expect(switchModalVisibility).to.be.not.called;
            wrapper.find(TouchableOpacity).at(1).props().onPress();
            expect(switchModalVisibility).to.be.calledOnce;
        });

        it("Should call switchModalVisibility callback when pressing the \"Confirmer\" button", () => {
            expect(confirmAnswerFunction).to.be.not.called;
            wrapper.find(TouchableOpacity).last().props().onPress();
            expect(confirmAnswerFunction).to.be.calledOnce;
        });
    });
});
