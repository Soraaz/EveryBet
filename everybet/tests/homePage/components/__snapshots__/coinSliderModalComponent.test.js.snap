// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinSliderModalComponent rendering Should render correctly the component 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000040",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 5,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "bold",
            "marginTop": 10,
            "textAlign": "center",
          }
        }
      >
        test
      </Text>
      <View
        style={
          Object {
            "borderBottomColor": "#000000",
            "borderBottomWidth": 1,
            "marginVertical": 10,
            "width": "50%",
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        Vous avez choisi la réponse :
      </Text>
      <Text
        style={
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "marginHorizontal": 10,
            "textAlign": "center",
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        Veuillez entrer le montant de betCoins
que vous voulez placer sur cette réponse :
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingLeft": 25,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "#707070",
                "borderRadius": 3,
                "borderWidth": 0.5,
                "height": 25,
                "marginTop": 10,
                "paddingHorizontal": 10,
                "width": "55%",
              },
              Object {
                "textAlign": "right",
              },
            ]
          }
          underlineColorAndroid="transparent"
          value="1"
        />
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/betcoin_icon.png",
            }
          }
          style={
            Object {
              "height": 25,
              "marginLeft": 5,
              "marginTop": 10,
              "width": 25,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FC8D8D",
              "borderRadius": 5,
              "flex": 1,
              "height": 31,
              "justifyContent": "center",
              "margin": 2,
              "marginLeft": 1,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginHorizontal": 10,
                "textAlign": "center",
              }
            }
          >
            Annuler
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#A6E48D",
              "borderRadius": 5,
              "flex": 1,
              "height": 31,
              "justifyContent": "center",
              "margin": 2,
              "marginRight": 1,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginHorizontal": 10,
                "textAlign": "center",
              }
            }
          >
            Confirmer
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
